applications/wg-easy/taskfiles/container.yml

```
@@ -0,0 +1,90 @@
+version: "3"
+
+# Development environment tasks
+tasks:
+  build-image:
+    desc: Build development Docker image
+    vars:
+      IMAGE_NAME: '{{.IMAGE_NAME}}'
+      IMAGE_TAG: '{{.IMAGE_TAG | default "latest"}}'
+      CONTAINERFILE: '{{.CONTAINERFILE | default "./container/Containerfile"}}'
+      BUILD_ARGS: '{{.BUILD_ARGS | default ""}}'
+    requires:
+      vars: [IMAGE_NAME, IMAGE_TAG, CONTAINERFILE]
+
+    cmds:
+      - docker build -t {{.IMAGE_NAME}} -f {{.CONTAINERFILE}} .
+
+  start:
+    desc: Start development container in background
+    silent: true
+    cmds:
+      - task: start-implementation
+
+  # Start development container in background.
+  # It's internal because it's used by shell and start tasks.
+  start-implementation:
+    desc: Start development container in background
+    silent: true
+    run: once
+    internal: true
+    vars:
+      IMAGE_NAME: '{{.IMAGE_NAME}}'
+      CONTAINER_NAME: '{{.CONTAINER_NAME}}'
+      IMAGE_TAG: '{{.IMAGE_TAG | default "latest"}}'
+    requires:
+      vars: [IMAGE_NAME, CONTAINER_NAME, REPLICATED_API_TOKEN]
+
+    status:
+      - docker ps | grep -q "{{.CONTAINER_NAME}}"
+    cmds:
+      - |
+        # Start container with host networking for kubectl port-forward compatibility
+        CONTAINER_ID=$(docker run --rm --name {{.CONTAINER_NAME}} -d \
+          -v $(pwd):/workspace \
+          -e HOME=/home/devuser \
+          -e USER=devuser \
+          -e REPLICATED_API_TOKEN={{ .REPLICATED_API_TOKEN }} \
+          -w /workspace \
+          {{.IMAGE_NAME}}:{{.IMAGE_TAG}} bash -c 'trap "exit" TERM; while :; do sleep 0.1; done')
```

```suggestion
          {{.IMAGE_NAME}}:{{.IMAGE_TAG}} bash -c 'trap "exit 0" TERM INT; sleep infinity & wait')
```

There's no need to process something every .1 seconds just sleep forever, exit on the termination commands

---

` & wait` was the bit I needed. I couldn't get `SIGTERM` to work with `sleep infinity` when running in the container